### PR TITLE
PSR-18: Move goal above the specification

### PR DESCRIPTION
**Pull request by kelunik**
I think that's the better order.

### CODE REVIEW

**Comment by kelunik** [comment]
No idea, TBH. Just had a look at some PSRs, the all didn't contain a goal.

File: proposed/http-client/http-client.md [hunk 1]
```diff
@@ -7,6 +7,16 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
...
 
+## Goal
+
+The goal of this PSR is to allow developers to create libraries decoupled from HTTP client
+implementations. This will make libraries more reusable as it reduces the number of
+dependencies and lowers the likelihood to get version conflicts.
+
+The second goal is that HTTP clients can be replaced as per the
+[Liskov substitutions principle][Liskov]. This means that all clients MUST behave in the
+same when sending a request.
```
Comment (kelunik) [inline proposed/http-client/http-client.md]: Actually, I think this paragraph can be dropped. It's a general principle and not specific to this PSR.

Comment (dbu) [reply proposed/http-client/http-client.md]: but for this PSR its particularly relevant. its the reason why we want to define eg 3xx behaviour and not leave defaults to the choice of the implementations.

Comment (kelunik) [reply proposed/http-client/http-client.md]: Yes, but e.g. the cache PSR also doesn't mention it. You could argue the same with cache misses there.

**Comment by stefanotorresi** [comment]
What is listed here as the main goal, is usually stated in the very first section of the spec, after the title and before the RFC2119 reference. Other than that, goals and non-goals are usually listed more extensively in the meta document.

**Pull request closed without merging**